src/python/WMCore/Storage/RucioFileCatalog.py

```diff
@@ -0,0 +1,236 @@
+    #now loop over elements, select the right one and fill lfn-to-pfn
+    for jsElement in jsElements:
+        #check to see if the storageSite and volume matchs with "site" and "volume" in storage.json
+        if jsElement['site'] == storageSite and jsElement['volume'] == volume: 
+            rfcInstance.preferredProtocol = protocol
+            #now loop over protocols to add all mappings (needed for chained rule cases)
+            for prot in jsElement['protocols']:
+                #check if prefix is in protocol block
+                if 'prefix' in prot.keys():
+                    #lfn-to-pfn
+                    match = '/(.*)' #match all
+                    result = prot['prefix']+'/$1'
+                    #prefix case should not be chained
+                    chain = None 
+                    rfcInstance.addMapping(str(prot['protocol']), str(match), str(result), chain, 'lfn-to-pfn')
+                    #pfn-to-lfn
+                    match = prot['prefix']+'/(.*)'
+                    result = '/$1'
+                    rfcInstance.addMapping(str(prot['protocol']), str(match), str(result), chain, 'pfn-to-lfn')
+                #here is rules  
+                else:
+                    #loop over rules
+                    for rule in prot['rules']:
+                        match = rule['lfn']
+                        result = rule['pfn']
+                        chain = rule['chain'] if 'chain' in rule.keys() else None
+                        rfcInstance.addMapping(str(prot['protocol']), str(match), str(result), chain, 'lfn-to-pfn')
+                        #pfn-to-lfn: not sure about this!!!
+                        match = rule['pfn'].replace('$1','(.*)')
+                        result = rule['lfn'].replace('/+','/').replace('^/','/')
+                        #now replace anything inside () with $1, for example (.*) --> $1, (store/.*) --> $1
+                        result = re.sub('\(.*\)','$1',result)
+                        rfcInstance.addMapping(str(prot['protocol']), str(match), str(result), chain, 'pfn-to-lfn')
+    
+    return rfcInstance
+
+def rseName(currentSite,currentSubsite,storageSite,volume):
+    """
+    Return Rucio storage element name, for example https://gitlab.cern.ch/SITECONF/T1_DE_KIT/-/blob/master/storage.json?ref_type=heads#L39
+    :currentSite is the site where jobs are executing
+    :currentSubsite is the sub site if jobs are running here
+    :storageSite is the site for storage
+    :volume is the volume name, for example https://gitlab.cern.ch/SITECONF/T1_DE_KIT/-/blob/master/storage.json?ref_type=heads#L3
+    """
+    rse = None
+    storageJsonName = storageJsonPath(currentSite,currentSubsite,storageSite)
+    try:
+        with open(storageJsonName,encoding="utf-8") as jsonFile:
+            jsElements = json.load(jsonFile)
+    except Exception as ex:
+        msg = "RucioFileCatalog.py:rseName_rucio() Error reading storage.json: %s\n" % storageJsonName 
```

Comment:
This error message seems to be outdated with the reference to "rseName_rucio"

Reply:
Done